src/server/cmd/pachd/main.go

```diff
@@ -747,10 +736,16 @@ func logGRPCServerSetup(name string, f func() error) (retErr error) {
 	log.Printf("started setting up %v GRPC Server", name)
 	defer func() {
 		if retErr != nil {
-			log.Printf("errored setting up %v GRPC Server", name)
+			retErr = fmt.Errorf("error setting up %v GRPC Server: %v", name, retErr)
 		} else {
 			log.Printf("finished setting up %v GRPC Server", name)
 		}
 	}()
 	return f()
 }
+
+func waitForError(name string, errChan chan error, f func() error) {
+	if err := f(); err != http.ErrServerClosed {
+		errChan <- fmt.Errorf("error in %v: %v", name, f())
+	}
+}
```

Comment:
So this means that if multiple things error this function will hang, but that's ok because the first error will cause the program to crash anyways, so blocking a few goros doesn't matter. Is that all accurate?

Reply:
That is correct. The program will log the first error and exit. It did not really make sense to have some notion of cleaning up goroutines after the first error if the program is going to exit anyways.